### PR TITLE
Attempt to clarify Filter Conditions for voiding statements

### PR DESCRIPTION
**Pull request by garemoko**
Attempt at addressing #632

### CODE REVIEW

File: xAPI.md [hunk 1]
```diff
@@ -3226,10 +3226,16 @@ include attachment raw data and MUST report application/json.
 
 ###### Filter Conditions for StatementRefs 
 
-For filter parameters which are not time or sequence based (that is, other than
-since, until, or limit), Statements which target another Statement (by using a StatementRef
+This section outlines rules by which additional Statements are considered to meet the filter coniditions even
+if they do not match the original query's filter parameters but instead target a Statement which does.
```
Comment (fugu13) [inline xAPI.md]: For this sentence to be correct, we have to introduce a second meaning of matching the filter parameters below (instead of just talking about matching some of the filter parameters as before). That's a lot of extra complication.

File: xAPI.md [hunk 2]
```diff
@@ -3226,10 +3226,16 @@ include attachment raw data and MUST report application/json.
...
-since, until, or limit), Statements which target another Statement (by using a StatementRef
+This section outlines rules by which additional Statements are considered to meet the filter coniditions even
+if they do not match the original query's filter parameters but instead target a Statement which does. 
+These rules **do not** apply when retrieving a single Statement with statementId or voidedStatementId.
+
+For filter parameters which are not time or sequence based (that is, other than since, until, or limit), 
+Statements which target another Statement (by using a StatementRef
 as the Object of the Statement) will meet the filter condition if the targeted Statement meets 
-the condition. The time and sequence based parameters MUST still be applied to the Statement 
+the condition of that filter parameter.
```
Comment (fugu13) [inline xAPI.md]: in the previous text we used "meet the filter condition/meets the condition" intentionally, as the rule applies to itself. This changes it to "meet the filter condition/meets the condition of that filter parameter". I'm not actually sure what the second bit means, but this makes it hard to read recursively.

Actually, I think that by changing it to talk about singular filter parameters, this text actually breaks the intended meaning. The previous wording is dense and a bit hard to follow, but exact, and I don't want to replace it with anything that isn't equally exact (as this is a pretty important part of the spec).

File: xAPI.md (hunk 1 above)
Comment (garemoko) [inline xAPI.md]: Yeh, I was trying to summarise simply for an introduction. Maybe I can make it more generic. 
How about:

``` This section outlines rules by which Statements targeting other Statements can sometimes be considered to meet the filter conditions of a query even if they do not match the original query's filter parameters. ```

File: xAPI.md (hunk 2 above)
Comment (garemoko) [inline xAPI.md]: Point taken. I will revert the addition of "of that filter parameter" and change "the condition" to "the filter condition".

**Comment by garemoko** [comment]
@fugu13 those this address your concerns in the outdated diff?

**Comment by andyjohnson** [comment]
First paragraph change "apply when retrieving a single Statement with statementId or voidedStatementId" to "apply when retrieving a single Statement using statementId or voidedStatementId as search parameters"

Since, until, and limit in double quotes.  Probably statementId and voidedStatementId too, but I could see being lax on these as they are clearly not going to be confused as normal language.

Otherwise +1

File: xAPI.md [hunk 3]
```diff
@@ -3226,10 +3226,21 @@ include attachment raw data and MUST report application/json.
...
 
-For filter parameters which are not time or sequence based (that is, other than
-since, until, or limit), Statements which target another Statement (by using a StatementRef
+This section outlines rules by which Statements targeting other Statements can sometimes be considered to 
+meet the filter conditions of a query even if they do not match the original query's filter parameters. 
+These rules **do not** apply when retrieving a single Statement using "statementId" or "voidedStatementId" query 
+parameters.
+
+'Targeting Statements' means that one Statement (the targeting Statement) includes the Statement Id of another
+Statement (the targeted Statement) as a Statemnent Reference either as the object of the Statement or within the
```
Comment (fugu13) [inline xAPI.md]: minor typo, "Statemnent"

File: xAPI.md [hunk 4]
```diff
@@ -3226,10 +3226,21 @@ include attachment raw data and MUST report application/json.
...
-For filter parameters which are not time or sequence based (that is, other than
-since, until, or limit), Statements which target another Statement (by using a StatementRef
+This section outlines rules by which Statements targeting other Statements can sometimes be considered to 
+meet the filter conditions of a query even if they do not match the original query's filter parameters. 
+These rules **do not** apply when retrieving a single Statement using "statementId" or "voidedStatementId" query 
+parameters.
+
+'Targeting Statements' means that one Statement (the targeting Statement) includes the Statement Id of another
+Statement (the targeted Statement) as a Statemnent Reference either as the object of the Statement or within the 
+"statement" property of the Context.
```
Comment (fugu13) [inline xAPI.md]: This seems to be an expansion, taken with some parts below. In the next sentence you see that it only applied (applies?) to ones in the object of the statement, not ones in context... but then later on line 3284 we refer to the generalized targeting definition, not the definition restricted to just the object. I believe we need to remove the context mention.

**Comment by andyjohnson** [comment]
There is a stray "n" in this paragraph in one of the "statements":

'Targeting Statements' means that one Statement (the targeting Statement) includes the Statement Id of another Statement (the targeted Statement) as a Statemnent Reference either as the object of the Statement or within the "statement" property of the Context.

**Comment by garemoko** [comment]
This has already been merged; suggest raising this as a separate issue or just making a quick PR.

**Comment by andyjohnson** [comment]
Consensus on 4/15/15 to revert to only use the Statement Refs, mostly for the voided cases.  Recommend calling out context "statement"  as not part of this.
